tsfc/fiatinterface.py

```diff
@@ -178,6 +173,15 @@ def _(element, vector_is_mixed):
         # Real element is just DG0
         cell = element.cell()
         return create_element(ufl.FiniteElement("DG", cell, 0), vector_is_mixed)
+    if element.family() == "Quadrature":
+        # Sneaky import from FFC
+        from ffc.quadratureelement import QuadratureElement
+        ffc_element = QuadratureElement(element)
+
+        def tabulate(self, order, points, entity):
+            return QuadratureElement.tabulate(self, order, points)
+        ffc_element.tabulate = types.MethodType(tabulate, ffc_element)
+        return ffc_element
```

Comment:
Not quite sure which one of these two approaches is better.

Perhaps we should just move all FIAT-like elements from FFC to FIAT. If we consider all of FFC, FIAT and TSFC, the current situation is a bit chaotic:

- Both FFC and FIAT have an `EnrichedElement`. FFC uses its own, TSFC uses the one from FIAT.
- Both FFC and TSFC have a `MixedElement`. Both using their own, however, due to form splitting, coefficient splitting and `finat.TensorFiniteElement`, Firedrake never actually uses the TSFC `MixedElement`, so the TSFC copy is only there for the sake of FEniCS.
- FFC has a `QuadratureElement`. FInAT has a smarter `QuadratureElement` which shall be satisfactory for Firedrake in virtually all cases. (Unless someone wants to "enrich" quadrature elements with other elements.) Because FEniCS does not practice form splitting and coefficient splitting, there many more cases in FEniCS when TSFC needs a FIAT-like `QuadratureElement`  which TSFC sneakily imports from FFC.